protocols/index/contracts/Index.sol

```diff
@@ -13,207 +13,196 @@
-    _listEntries[HEAD] = Entry(HEAD, HEAD, 0, bytes32(0));
+    // Create initial entry.
+    _entries[HEAD] = Entry(bytes32(0), 0, HEAD, HEAD);
   }
 
   /**
-    * @notice Set an Entry
-    *
-    * @param _user The account
+    * @notice Set a Locator
+    * @param _identifier address
     * @param _score uint256
     * @param _locator bytes32
     */
-  function setEntry(
-    address _user,
+  function setLocator(
+    address _identifier,
     uint256 _score,
     bytes32 _locator
   ) external onlyOwner {
 
-    require(!hasEntry(_user), "ENTRY_ALREADY_EXISTS");
+    // Ensure the entry does not already exist.
+    require(!hasEntry(_identifier), "ENTRY_ALREADY_EXISTS");
 
-    // Find the first user who has a lower stake, and insert before them
-    address nextUser = findPosition(_score);
+    // Find the first entry with a lower score.
+    address nextEntry = getEntryLowerThan(_score);
 
-    // Link the newUser into place.
-    address prevUser = _listEntries[nextUser].prev;
+    // Link the new entry between previous and next.
+    address prevEntry = _entries[nextEntry].prev;
+    _entries[prevEntry].next = _identifier;
+    _entries[nextEntry].prev = _identifier;
+    _entries[_identifier] = Entry(_locator, _score, prevEntry, nextEntry);
 
-    _listEntries[prevUser].next = _user;
-    _listEntries[nextUser].prev = _user;
-    _listEntries[_user] = Entry(prevUser, nextUser, _score, _locator);
-
-    // Increment the length of the linked list if successful.
+    // Increment the index length.
     length = length + 1;
-
-    emit SetEntry(_score, _user, _locator);
+    emit SetLocator(_identifier, _score, _locator);
   }
 
   /**
-    * @notice Unset an Entry
-    * @param _user address
+    * @notice Unset a Locator
+    * @param _identifier address
     * @return bool return true on success
     */
-  function unsetEntry(
-    address _user
+  function unsetLocator(
+    address _identifier
   ) external onlyOwner returns (bool) {
 
-    // Ensure the _user is in the linked list.
-    if (!hasEntry(_user)) {
-      return false;
-    }
+    // Ensure the entry exists.
+    require(hasEntry(_identifier), "ENTRY_DOES_NOT_EXIST");
 
-    // Link its neighbors together.
-    address prevUser = _listEntries[_user].prev;
-    address nextUser = _listEntries[_user].next;
+    // Link the previous and next entries together.
+    address prevUser = _entries[_identifier].prev;
+    address nextUser = _entries[_identifier].next;
+    _entries[prevUser].next = nextUser;
+    _entries[nextUser].prev = prevUser;
 
-    _listEntries[prevUser].next = nextUser;
-    _listEntries[nextUser].prev = prevUser;
+    // Delete entry from the index.
+    delete _entries[_identifier];
 
-    // Delete user from the list.
-    delete _listEntries[_user];
-
-    // Decrement the length of the linked list.
+    // Decrement the index length.
     length = length - 1;
 
-    emit UnsetEntry(_user);
+    emit UnsetLocator(_identifier);
     return true;
```

Comment:
Do we still need it to return a bool? Now that a failure throws in a require (line 109), it feels like the fact it hasn't thrown is enough to know it passed?